operator/src/test/java/io/strimzi/kafka/access/KafkaAccessReconcilerTest.java

```diff
@@ -523,4 +525,56 @@ void testReconcileInvalidUserReference(KafkaUserReference kafkaUserReference) {
         assertThat(readyCondition.getMessage()).isEqualTo("User kind must be KafkaUser and apiGroup must be kafka.strimzi.io");
         assertThat(readyCondition.getReason()).isEqualTo("InvalidUserKind");
     }
+
+    @Test
+    @DisplayName("Reconciler should use a user-provided secret name but also delete the old secret if it changes")
+    void testReconcileWithUserProvidedSecretAndTestDeleteSecretWithNameChange() {
+        final Kafka kafka = ResourceProvider.getKafka(
+                KAFKA_NAME,
+                KAFKA_NAMESPACE,
+                List.of(ResourceProvider.getListener(LISTENER_1, KafkaListenerType.INTERNAL, false)),
+                List.of(ResourceProvider.getListenerStatus(LISTENER_1, BOOTSTRAP_HOST, BOOTSTRAP_PORT_9092))
+        );
+        Crds.kafkaOperation(client).inNamespace(KAFKA_NAMESPACE).resource(kafka).create();
+
+        final KafkaReference kafkaReference = ResourceProvider.getKafkaReference(KAFKA_NAME, KAFKA_NAMESPACE);
+        final KafkaAccess kafkaAccess = ResourceProvider.getKafkaAccess(NAME, NAMESPACE, kafkaReference);
+
+        kafkaAccess.getSpec().setSecretName(USER_PROVIDED_SECRET_NAME);
+        client.resources(KafkaAccess.class).resource(kafkaAccess).create();
+        client.resources(KafkaAccess.class).inNamespace(NAMESPACE).withName(NAME).waitUntilCondition(updatedKafkaAccess -> {
+            final Optional<String> bindingName = Optional.ofNullable(updatedKafkaAccess)
+                    .map(KafkaAccess::getStatus)
+                    .map(KafkaAccessStatus::getBinding)
+                    .map(BindingStatus::getName);
+            return bindingName.isPresent() && USER_PROVIDED_SECRET_NAME.equals(bindingName.get());
+        }, TEST_TIMEOUT, TimeUnit.MILLISECONDS);
+
+        Secret oldSecretBeforeRename = client.secrets().inNamespace(NAMESPACE).withName(USER_PROVIDED_SECRET_NAME).get();
+        assertThat(oldSecretBeforeRename).isNotNull();
+        assertThat(oldSecretBeforeRename.getType()).isEqualTo("servicebinding.io/kafka");
+        assertThat(oldSecretBeforeRename.getMetadata().getName()).isEqualTo(USER_PROVIDED_SECRET_NAME);
```

Comment:
This assert is not needed, since we know this is the name of the Secret as we used `client.secrets().withName` to fetch it

Reply:
Yeah +1